httpserver/idp_handlers.go

```diff
@@ -0,0 +1,490 @@
+        .form-group {
+            margin-bottom: 1rem;
+        }
+        label {
+            display: block;
+            margin-bottom: 0.5rem;
+            color: #555;
+        }
+        input[type="text"],
+        input[type="password"] {
+            width: 100%;
+            padding: 0.5rem;
+            border: 1px solid #ddd;
+            border-radius: 4px;
+            box-sizing: border-box;
+        }
+        button {
+            width: 100%;
+            padding: 0.75rem;
+            background-color: #007bff;
+            color: white;
+            border: none;
+            border-radius: 4px;
+            cursor: pointer;
+            font-size: 1rem;
+        }
+        button:hover {
+            background-color: #0056b3;
+        }
+        .error {
+            color: #d9534f;
+            margin-bottom: 1rem;
+        }
+    </style>
+</head>
+<body>
+    <div class="login-container">
+        <h2>HTCondor IDP Login</h2>
+        <form method="POST" action="/idp/login">
+            <input type="hidden" name="redirect_uri" value="` + redirectURI + `">
+            <div class="form-group">
+                <label for="username">Username:</label>
+                <input type="text" id="username" name="username" required autofocus>
+            </div>
+            <div class="form-group">
+                <label for="password">Password:</label>
+                <input type="password" id="password" name="password" required>
+            </div>
+            <button type="submit">Login</button>
+        </form>
+    </div>
+</body>
+</html>`
+
+	w.Header().Set("Content-Type", "text/html")
+	w.WriteHeader(http.StatusOK)
+	_, _ = w.Write([]byte(html))
+}
+
+// handleIDPLoginSubmit handles login form submission
+func (s *Server) handleIDPLoginSubmit(w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+		s.writeError(w, http.StatusBadRequest, "Failed to parse form")
+		return
+	}
+
+	username := r.FormValue("username")
+	password := r.FormValue("password")
+	redirectURI := r.FormValue("redirect_uri")
+
+	if username == "" || password == "" {
+		s.writeError(w, http.StatusBadRequest, "Username and password required")
+		return
+	}
+
+	// Authenticate user
+	ctx := r.Context()
+	if err := s.idpProvider.storage.AuthenticateUser(ctx, username, password); err != nil {
+		s.logger.Warn(logging.DestinationHTTP, "IDP authentication failed", "username", username)
+		s.writeError(w, http.StatusUnauthorized, "Invalid credentials")
+		return
+	}
+
+	s.logger.Info(logging.DestinationHTTP, "IDP user authenticated", "username", username)
+
+	// Create session for authenticated user
+	// Store username in cookie or session store (using simple cookie for now)
+	// TODO: Set Secure flag to true in production with HTTPS
+	http.SetCookie(w, &http.Cookie{
+		Name:     "idp_session",
+		Value:    username,
+		Path:     "/",
+		HttpOnly: true,
+		Secure:   false, // Set to true when using HTTPS in production
```

Comment:
Always set to true.

For demo mode, generate a self-signed cert if certs aren't provided.  This should allow secure mode to be used.